src/snac/server/database/SQL.php

```diff
@@ -4034,6 +4034,51 @@ public function selectTerm($termID)
     }
 
 
+    /**
+     * Get a single vocabulary record by Type and Value
+     *
+     * @param string $value The value of a vocabulary term
+     *
+     * @param string $type The type of a vocabulary term
+     *
+     * @return string[] A list with keys: id, type, value, uri, description
+     *
+     */
+    public function selectTermByValueAndType($value, $type)
+    {
+        $qq = 's_term_by_value';
+        $this->sdb->prepare($qq,
+                            'select
+                            id, type, value, uri, description
+                            from vocabulary where value=$1 and type=$2');
+        $result = $this->sdb->execute($qq, array($value, $type));
+        $row = $this->sdb->fetchrow($result);
```

Comment:
What does this do if a matching term isn't found?  Is row just null?  I'm curious if we think there could be two terms that match, but I'm doubtful that would be the case.